logging/src/commonMain/kotlin/io/spine/logging/LoggingFactory.kt

```diff
@@ -46,6 +46,28 @@ public expect object LoggingFactory {
      * If the domain is not specified, returns [LoggingDomain.noOp].
      */
     public fun loggingDomainOf(cls: KClass<*>): LoggingDomain
+
+    /**
+     * Creates a key for a single piece of metadata. If metadata is set more than once
+     * using this key for the same log statement, the last set value will be the one used, and
+     * other values will be ignored (although callers should never rely on this behavior).
+     *
+     * Key instances behave like singletons, and two key instances with the same label will still
+     * be considered distinct. The recommended approach is to always assign [MetadataKey]
+     * instances to static final constants.
+     */
+    public fun <T: Any> singleMetadataKey(label: String, type: KClass<T>): MetadataKey<T>
```

Comment:
I would rename `type: KClass<T>` to `valueType: KClass<T>` because now it reads as `type of the key`. Or document type parameter `T`, renaming the method argument to `valueClass`.

The same for the method below.